### PR TITLE
Added new migration for special characters

### PR DESCRIPTION
**Pull request by Trial97**

### CODE REVIEW

**Review by Ryex — APPROVED** [review]

**Review by DioEgizio — APPROVED** [review]

File: tests/INIFile_test.cpp [hunk 1]
```diff
@@ -93,11 +98,69 @@ PreLaunchCommand="$INST_JAVA" -jar packwiz-installer-bootstrap.jar link
...
         QCOMPARE(f2.get("PreLaunchCommand", "NOT SET").toString(), "\"$INST_JAVA\" -jar packwiz-installer-bootstrap.jar link");
-        QCOMPARE(f2.get("ConfigVersion", "NOT SET").toString(), "1.1");
+        QCOMPARE(f2.get("Wrapperommand", "NOT SET").toString(), "\"$INST_JAVA\" -jar packwiz-installer-bootstrap.jar link =");
+        QCOMPARE(f2.get("ConfigVersion", "NOT SET").toString(), "1.2");
+    }
+
+    void test_SaveAleardyExistingFileWithSpecialChars()
+    {
+        QString fileName = "test_SaveAleardyExistingFileWithSpecialChars.ini";
+        FS::deletePath(fileName);  // just to clean the previous test run
```
Comment (Scrumplex) [inline tests/INIFile_test.cpp]: maybe use `QTemporaryFile` instead?

File: tests/INIFile_test.cpp [hunk 2]
```diff
@@ -93,11 +98,69 @@ PreLaunchCommand="$INST_JAVA" -jar packwiz-installer-bootstrap.jar link
...
+        INIFile f2;
+        f2.loadFile(fileName);
+        for (auto key : settings.allKeys())
+            QCOMPARE(f2.get(key, "NOT SET").toString(), settings.value(key).toString());
+        QCOMPARE(f2.get("ConfigVersion", "NOT SET").toString(), "1.2");
+    }
+
+    void test_SaveAleardyExistingFileWithSpecialCharsV1()
+    {
+        QString fileName = "test_SaveAleardyExistingFileWithSpecialCharsV1.ini";
```
Comment (Scrumplex) [inline tests/INIFile_test.cpp]: same here

File: tests/INIFile_test.cpp [hunk 3]
```diff
@@ -93,11 +98,69 @@ PreLaunchCommand="$INST_JAVA" -jar packwiz-installer-bootstrap.jar link
...
+            QCOMPARE(f1.get(key, "NOT SET").toString(), settings.value(key).toString());
+        f1.saveFile(fileName);
+        INIFile f2;
+        f2.loadFile(fileName);
+        for (auto key : settings.allKeys())
+            QCOMPARE(f2.get(key, "NOT SET").toString(), settings.value(key).toString());
+        QCOMPARE(f2.get("ConfigVersion", "NOT SET").toString(), "1.2");
+    }
+
+    void test_SaveAleardyExistingFileWithSpecialCharsV1()
```
Comment (Scrumplex) [inline tests/INIFile_test.cpp]:
```suggestion
    void test_SaveAlreadyExistingFileWithSpecialCharsV1()
```

File: tests/INIFile_test.cpp [hunk 4]
```diff
@@ -93,11 +98,69 @@ PreLaunchCommand="$INST_JAVA" -jar packwiz-installer-bootstrap.jar link
...
         f1.saveFile(fileName);
         INIFile f2;
         f2.loadFile(fileName);
         QCOMPARE(f2.get("PreLaunchCommand", "NOT SET").toString(), "\"$INST_JAVA\" -jar packwiz-installer-bootstrap.jar link");
-        QCOMPARE(f2.get("ConfigVersion", "NOT SET").toString(), "1.1");
+        QCOMPARE(f2.get("Wrapperommand", "NOT SET").toString(), "\"$INST_JAVA\" -jar packwiz-installer-bootstrap.jar link =");
+        QCOMPARE(f2.get("ConfigVersion", "NOT SET").toString(), "1.2");
+    }
+
+    void test_SaveAleardyExistingFileWithSpecialChars()
```
Comment (Scrumplex) [inline tests/INIFile_test.cpp]:
```suggestion
    void test_SaveAlreadyExistingFileWithSpecialChars()
```

File: launcher/settings/INIFile.cpp [hunk 5]
```diff
@@ -97,6 +97,20 @@ QString unescape(QString orig)
     }
     return out;
 }
+
+QString unquete(QString str)
```
Comment (Scrumplex) [inline launcher/settings/INIFile.cpp]:
```suggestion
QString unquote(QString str)
```

File: launcher/settings/INIFile.cpp [hunk 6]
```diff
@@ -124,7 +138,7 @@ bool parseOldFileFormat(QIODevice& device, QSettings::SettingsMap& map)
         QString key = line.left(eqPos).trimmed();
         QString valueStr = line.right(line.length() - eqPos - 1).trimmed();
 
-        valueStr = unescape(valueStr);
+        valueStr = unquete(unescape(valueStr));
```
Comment (Scrumplex) [inline launcher/settings/INIFile.cpp]:
```suggestion
        valueStr = unquote(unescape(valueStr));
```

**Comment by Trial97** [comment]
Regarding the temp file stuff: I used it on UNIX systems but for windows, I used the old approach: FS:deletePath (it failed in the testing without any output)

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**